### ISSUE TITLE
Unit Test Issues

### ISSUE THREAD

**Issue by autolycus** [ethereum/aleth#162]
Hi, 

I'm attempting to build and run the unit tests. I've built headless and I can run the 'eth' executable. I've cloned the 'tests' repo. 

However, when I run some of the unit tests, I seem to encounter issues. For example: 

```
$> pwd
~/src/cpp-ethereum/test
$> testeth --run_test=basic_tests
Running 1 test case...
*** [ 14:54:38 | main ] Testing Hex-Prefix-Encode...
*** [ 14:54:38 | main ] nz,even,open
*** [ 14:54:38 | main ] nz,even,term
*** [ 14:54:38 | main ] nz,odd,open
*** [ 14:54:38 | main ] nz,odd,term
*** [ 14:54:38 | main ] z,even,open
*** [ 14:54:38 | main ] z,even,term
*** [ 14:54:38 | main ] z,odd,open
*** [ 14:54:38 | main ] z,odd,term
*** [ 14:54:38 | main ] zz,even,open
*** [ 14:54:38 | main ] zz,even,term
*** [ 14:54:38 | main ] zz,odd,open
*** [ 14:54:38 | main ] zz,odd,term
*** [ 14:54:38 | main ] Testing RLP...
*** [ 14:54:38 | main ] bigint
*** [ 14:54:38 | main ] emptylist
*** [ 14:54:38 | main ] emptystring
*** [ 14:54:38 | main ] listsoflists
*** [ 14:54:38 | main ] longstring
*** [ 14:54:38 | main ] mediumint
*** [ 14:54:38 | main ] multilist
*** [ 14:54:38 | main ] shortstring
*** [ 14:54:38 | main ] smallint
*** [ 14:54:38 | main ] stringlist
*** [ 14:54:38 | main ] zero
*** [ 14:54:38 | main ] Testing Trie...
*** [ 14:54:38 | main ] basic
*** [ 14:54:38 | main ] basic1
*** [ 14:54:38 | main ] basic2
*** [ 14:54:38 | main ] beprefix
*** [ 14:54:38 | main ] doprefix
*** [ 14:54:38 | main ] multiprefix
*** [ 14:54:38 | main ] Testing VM...
*** [ 14:54:38 | main ] arith
unknown location(0): fatal error in "basic_tests": std::runtime_error: value type is 6 not 2
~/src/cpp-ethereum/test/vm.cpp(331): last checkpoint

*** 1 failure detected in test suite "EthereumTests"
```

When I use gdb, I can see the the VM test code pulls the vmtests.json file in as a string. 
In the doTests() vm.cpp:338, There is the following code where the exception gets thrown:

```
            for (auto i: o["exec"].get_array())
            {
                Value_type t = i.type(); 
>>>>>>>>>       fev.importExec(i.get_obj());
                output = vm.go(fev).toBytes();
            }
```

Basically, the object in the "exec" json array don't have a "caller" parameter defined in the object and this causes an exception. 

Who is maintaining the unit tests? From what I can tell, the unit tests seem to be a mixture of a home grown solution and several that are using boost's unit testing framework.

Is anyone opposed to refactoring some of these tests to use boost's unit testing framework only?
